connectivity/manifests/allow-all-except-world.yaml

```diff
@@ -14,6 +14,16 @@ spec:
     - kube-apiserver
   - toEndpoints:
     - {}
+  # When node-local-dns is deployed with local IP,
```

Comment:
Why not narrow this down further  -
```
- matchExpressions:
          - { key: 'k8s-app', operator: In, values: [ "node-local-dns", "nodelocaldns" ] }
          - { key: 'io.kubernetes.pod.namespace', operator: In, values: [ "kube-system" ] }
```

Reply:
Yeah, I added and tested there wasn't any issue. thanks.

Reply:
The new change caused a failure for `Multicluster` test. You can find the logs [here](https://github.com/cilium/cilium-cli/actions/runs/3882219651/jobs/6622177117#step:19:10935).